eng/pipelines/templates/jobs/ci.yml

```diff
@@ -226,17 +228,39 @@ jobs:
           BaselineFilePath: $(Build.sourcesdirectory)/eng/dotnet.gdnbaselines
           ServiceDirectory: ${{ parameters.ServiceDirectory }}
 
-  - template: /eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml
-    parameters:
-      JobTemplatePath: /eng/pipelines/templates/jobs/ci.tests.yml
-      MatrixConfigs: ${{ parameters.MatrixConfigs }}
-      MatrixFilters: ${{ parameters.MatrixFilters }}
-      MatrixReplace: ${{ parameters.MatrixReplace }}
-      CloudConfig:
-        Cloud: Public
-      AdditionalParameters:
-        SDKType: ${{ parameters.SDKType }}
-        ServiceToTest: ${{ coalesce(parameters.ServiceToTest, parameters.ServiceDirectory) }}
-        ServiceDirectory: ${{ parameters.ServiceDirectory }}
-        TestSetupSteps: ${{ parameters.TestSetupSteps }}
-        TestTimeoutInMinutes: ${{ parameters.TestTimeoutInMinutes }}
+        ${{ if ne(parameters.TestDependsOnDependency, 'not-specified') }}:
+          - template: /eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml
+            parameters:
+              JobTemplatePath: /eng/pipelines/templates/jobs/ci.tests.yml
+              MatrixConfigs: ${{ parameters.MatrixConfigs }}
+              MatrixFilters: ${{ parameters.MatrixFilters }}
+              MatrixReplace: ${{ parameters.MatrixReplace }}
+              CloudConfig:
+                Cloud: public
+              SparseCheckoutPaths: 
+                - "!/sdk/*/**/SessionRecords"
```

Comment:
I think we can include this in both places. Also shouldn't this have a trailing `/*` based on your perf findings here: https://github.com/Azure/azure-sdk-for-net/pull/29337/files#diff-95b8b6f25a62cf306705ecfc7a4d63b677351a6cfcac5c7d5f50257ba78d6c25R119